sim_pipeline/Deflector/deflector_base.py

```diff
@@ -0,0 +1,200 @@
+from abc import ABC, abstractmethod
+
+class DeflectorBase(ABC):
+    """
+    base class with functions all deflector classes must have to be able to render populations
+
+    """
+    def __init__(self, cosmo, sky_area):
+        """
+
+        :param cosmo: cosmology
+        :type cosmo: ~astropy.cosmology class
+        :param sky_area: Sky area over which galaxies are sampled. Must be in units of solid angle.
+        :type sky_area: `~astropy.units.Quantity`
+        """
+        self._cosmo = cosmo
+        self._sky_area = sky_area
+
+
+    @abstractmethod
+    def deflector_position(self):
+        """
+        center of the deflector position
+
+        :return: [x_pox, y_pos] in arc seconds
+        """
+        pass
+
+    @abstractmethod
+    def source_position(self):
+        """
+        source position, either the center of the extended source or the point source. If not present from the cataloge,
+        it is drawn uniform within the circle of the test area centered on the lens
+
+        :return: [x_pos, y_pos]
+        """
+
+    @abstractmethod
+    def image_positions(self):
+        """
+        returns image positions by solving the lens equation, these are either the centers of the extended source, or
+        the point sources in case of (added) point-like sources, such as quasars or SNe.
+
+        :return: x-pos, y-pos
```

Comment:
do we want to specify that the positions are in arrays (as they can be multiple images)

Reply:
Good idea - updated!